app/models/katello/glue/candlepin/pool.rb

```diff
@@ -34,11 +34,11 @@ def get_for_owner(organization)
       end
 
       def import_pool(cp_pool_id, index_hosts = true)
-        pool = nil
+        json = candlepin_data(cp_pool_id)
         ::Katello::Util::Support.active_record_retry do
-          pool = Katello::Pool.where(:cp_id => cp_pool_id).first_or_create
+          pool = Katello::Pool.where(:cp_id => cp_pool_id, :organization => Organization.find_by(:label => json['owner']['key'])).first_or_create
+          pool.import_data(index_hosts, json)
```

Comment:
the root cause was that we have a race condition where these things are happening at the same time:

a)  app/models/katello/glue/candlepin/pool.rb#import_pool
```Pool.where(:cp_id => cp_id, :organization_id => org_id).first_or_create```
b)    app/models/katello/glue/candlepin/candlepin_object.rb
```
pool = Pool.where(:cp_id => cp_id).first_or_create 
pool.update_attributes(:organization_id => org_id)
```

If the order is just right, the find() will fail, but then the create() will fail.  And even repeating this 3 times, it can fail 3 times between the first_or_create and update_attributes in b)

Reply:
:+1: